### PR TITLE
fix cronjob creation with irrelevant config change

### PR DESCRIPTION
**Pull request by pradeepitm12**
observed behaviour :
  crojob get recreated with additon/update in the prune config
  annotaions, but it also gets recreated even if irrelevant change
  happen in any namespace.
fix:
  add config hash which is applied, and check it next reconcile it
  checks the last applied hash with the current config hash and
  only create cronjobs if oldhash not equal to current hash

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/common/prune.go [hunk 1]
```diff
@@ -120,7 +124,7 @@ func Prune(ctx context.Context, k kubernetes.Interface, tC *v1alpha1.TektonConfi
 	if pruningNamespaces.commonScheduleNs != nil && len(pruningNamespaces.commonScheduleNs) > 0 {
 		jobs := pruner.createAllJobContainers(pruningNamespaces.commonScheduleNs)
 		cronNameWithRandomSuffix := SimpleNameGenerator.RestrictLengthWithRandomSuffix(CronName)
-		if err := pruner.checkAndCreate(ctx, cronNameWithRandomSuffix, tC.Spec.TargetNamespace, tC.Spec.TargetNamespace, tC.Spec.Pruner.Schedule, jobs, ownerRef, pruneCronLabel); err != nil {
+		if err := pruner.checkAndCreate(ctx, cronNameWithRandomSuffix, tC.Spec.TargetNamespace, "", tC.Spec.Pruner.Schedule, jobs, ownerRef, pruneCronLabel, pruningNamespaces.configChanged); err != nil {
```
Comment (nikhil-thomas) [inline pkg/reconciler/common/prune.go:127]: the function signature has become too long. Can we capture some of these things in the `Pruner`struct.

In addition, can we break the `checkAndCreate`method into sub methods for improved redability

Comment (nikhil-thomas) [inline pkg/reconciler/common/prune.go:127]: `pruningNamespaces.configChanged`will this ever become `true`in this call.

how does `pruningNamespaces, err := prunableNamespaces(ctx, k, tC.Spec.Pruner)`change `configChanged`

**Comment by nikhil-thomas** [comment]
/approve
✅ lgtm

File: pkg/reconciler/common/prune.go (hunk 1 above)
Comment (pradeepitm12) [reply pkg/reconciler/common/prune.go:127]: I think you are right. Let me try this and update the pr.

Comment (pradeepitm12) [reply pkg/reconciler/common/prune.go]: Updated, reduced the func signature.
Also, update a bit of the function behavior.

**Comment by pradeepitm12** [comment]
cc: @savitaashture @sm43

**Pull request merged**